chain/chain/src/test_utils/kv_runtime.rs

```diff
@@ -1002,19 +1002,19 @@ impl RuntimeAdapter for KeyValueRuntime {
     fn validate_tx(
         &self,
         _shard_layout: &ShardLayout,
-        _transaction: &SignedTransaction,
+        transaction: SignedTransaction,
         _current_protocol_version: ProtocolVersion,
         _receiver_congestion_info: Option<ExtendedCongestionInfo>,
-    ) -> Result<(), InvalidTxError> {
-        Ok(())
+    ) -> Result<ValidatedTransaction, (InvalidTxError, SignedTransaction)> {
+        Ok(ValidatedTransaction::new(transaction).unwrap())
```

Comment:
This possibly changes the behaviour of `kv_runtime` as it now actually verifies things, but now that it does so, why not return `SignedTransaction` back on failure?

Reply:
> This possibly changes the behaviour of kv_runtime as it now actually verifies things

Indeed, the alternative was to have some "for test purpose" way of constructing a ValidatedTx and I felt like that is not really a good strategy.  We might have to do that depending on how many tests are failing and how hard it is to fix them.

> why not return SignedTransaction back on failure?

My thinking was that this is testing code and if something goes wrong we want to know as early as possible.  I don't have a strong feeling on this though.  pushing a commit to remove the unwrap.